components/user-manager/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/UserRealmProxy.java

```diff
@@ -666,7 +667,7 @@ private boolean isBulkImportSupported() throws UserAdminException {
      */
     private final String addPrimaryDomainIfNotExists(String userName) {
 
-        if ((userName.indexOf(UserCoreConstants.DOMAIN_SEPARATOR)) < 0) {
+        if ((userName.indexOf(UserCoreConstants.DOMAIN_SEPARATOR)) < 0 && StringUtils.isNotEmpty(userName)) {
```

Comment:
null check should comes first. ie,
if (StringUtils.isNotEmpty(userName) && (userName.indexOf(UserCoreConstants.DOMAIN_SEPARATOR)) < 0) {